### ISSUE TITLE
find a way to rebuild automatically the index

### ISSUE THREAD

**Issue by Shir0kamii** [tldr-pages/tldr#487]
Hi,

I've just discovered tldr, and I think it's a great tool.

One annoying thing though, is that the index wasn't up to date when I forked. It's normal for us developers to find out what wrong when we see a `vim.md` but cant do `tldr find vim`. But I think normal people wouldn't find help about vim in the first place if tldr don't find it.

For information purpose, I use tldr.py 0.2.0 and if a page isn't in the index, the client won't find it.

Should we add the index updating to the pre-commit hook ? I guess it's the best way to keep it up to date, regardless of the client. What do you guys think about this suggestion ?

**Comment by lord63** [comment]
> Should we add the index updating to the pre-commit hook ?

In fact, it was in our pre-commit hook and then we removed it.  You can take a look at this issue #336 . Currently we don't have a clean solution to solve this problem.

As for the [tldr.py](https://github.com/lord63/tldr.py) client for building the index, I've opened an issue here before: lord63/tldr.py#8

**Comment by ostera** [comment]
@Shir0kamii, we discussed and have a working solution here: https://github.com/tldr-pages/tldr/issues/336

Other than that, closing as this is related to a client not breaking their index/cache.

**Comment by Shir0kamii** [comment]
Okay, thank you two for the fast replies.

Please tell me if I misunderstood, but from what I read, the index.json in this repositories is not guarranted to be up to date ? If so, why not remove it ?

**Comment by ostera** [comment]
The updated index can be found on the tldr-pages.github.io repo, and you're
right, it should be removed from this one.

**Comment by dbrgn** [comment]
Same problem here, I relied on the `index.json` file to be up to date. But it actually contains entries that aren't present in the directroy structure.

Where exactly can I find an up to date index on https://tldr-pages.github.io/?

**Comment by igorshubovych** [comment]
tldr-pages.github.io/assets/index.json
